utils/frame/remote-externalities/src/lib.rs

```diff
@@ -112,13 +147,19 @@ impl Transport {
 
 impl From<String> for Transport {
 	fn from(uri: String) -> Self {
-		Transport::Uri(uri)
+		Transport { uri: Some(uri), remote_client: None }
+	}
+}
+
+impl From<&str> for Transport {
+	fn from(uri: &str) -> Self {
+		Transport { uri: Some(uri.to_string()), remote_client: None }
 	}
 }
 
 impl From<Arc<WsClient>> for Transport {
-	fn from(client: Arc<WsClient>) -> Self {
-		Transport::RemoteClient(client)
+	fn from(_: Arc<WsClient>) -> Self {
+		unimplemented!("we should not allow this. any client must be first built with a uri");
```

Comment:
question: Why? And why do we have this `impl` then?

Also, it seems like we can simplify `Transport`:
 - single constructor / conversion from `AsRef<str>`
 - deref / explicit downcasting to `Option<WsClient>`
 - no public fields
 - getter for the client (+ maybe just informational for uri / custom `Debug` impl)